src/js/components/modals/AppModalComponent.jsx

```diff
@@ -61,28 +61,21 @@ var AppModalComponent = React.createClass({
     Mousetrap.bind(["command+enter", "ctrl+enter"], () => this.handleSubmit());
 
     if (!this.hasInfoObject()) {
-      InfoStore.on(InfoEvents.CHANGE, this.onInfoChange);
-      InfoStore.on(InfoEvents.REQUEST_ERROR, this.onInfoChange);
+      InfoStore.once(InfoEvents.CHANGE, this.onInfoChange);
+      InfoStore.once(InfoEvents.REQUEST_ERROR, this.onInfoChange);
       InfoActions.requestInfo();
     }
   },
 
   componentWillUnmount: function () {
-    AppsStore.removeListener(AppsEvents.CREATE_APP,
-      this.onCreateApp);
+    AppsStore.removeListener(AppsEvents.CREATE_APP, this.onCreateApp);
     AppsStore.removeListener(AppsEvents.CREATE_APP_ERROR,
       this.onCreateAppError);
-    AppsStore.removeListener(AppsEvents.APPLY_APP,
-      this.onCreateApp);
-    AppsStore.removeListener(AppsEvents.APPLY_APP_ERROR,
-      this.onApplyAppError);
-    Mousetrap.unbind(["command+enter", "ctrl+enter"]);
-    this.removeInfoListeners();
-  },
-
-  removeInfoListeners: function () {
+    AppsStore.removeListener(AppsEvents.APPLY_APP, this.onCreateApp);
+    AppsStore.removeListener(AppsEvents.APPLY_APP_ERROR, this.onApplyAppError);
     InfoStore.removeListener(InfoEvents.CHANGE, this.onInfoChange);
     InfoStore.removeListener(InfoEvents.REQUEST_ERROR, this.onInfoChange);
```

Comment:
@aldipower I restored the `.removeListener` calls – with the `.once` in place we don't need to perform any checks so this should be good now.